lib/client.js

```diff
@@ -784,21 +783,24 @@ class Client extends events.EventEmitter {
             await this.#connect();
         }
 
-        return new Promise(async (resolve, reject) => {
-            try {
-                let statement = await this.rustClient.prepareStatement(query);
-                let parsedParams = parseParams(statement, params);
-                let rusty = await this.rustClient.executePrepared(
-                    statement,
-                    parsedParams,
-                );
-                resolve(new ResultSet(rusty));
-            } catch (e) {
-                reject(e);
-            }
+        return new Promise((resolve, reject) => {
+            this.#executePromise(resolve, reject, query, params);
         });
     }
 
+    async #executePromise(resolve, reject, query, params) {
+        try {
+            let statement = await this.rustClient.prepareStatement(query);
+            let parsedParams = parseParams(statement, params);
+            let rusty = await this.rustClient.executePrepared(
+                statement,
+                parsedParams,
+            );
+            resolve(new ResultSet(rusty));
+        } catch (e) {
+            reject(e);
+        }
+    }
```

Comment:
Justify the refactor here.

Reply:
ESLint doesn't allow previous version of the code:
https://eslint.org/docs/latest/rules/no-async-promise-executor

In the future we can consider doing it the opposite way (using async instead of promise).